dataproc/src/main/java/CreateCluster.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START create_cluster]
+import com.google.cloud.dataproc.v1.Cluster;
+import com.google.cloud.dataproc.v1.ClusterConfig;
+import com.google.cloud.dataproc.v1.ClusterControllerClient;
+import com.google.cloud.dataproc.v1.ClusterControllerSettings;
+import com.google.cloud.dataproc.v1.InstanceGroupConfig;
+import java.io.IOException;
+
+public class CreateCluster {
+
+  public void createCluster(String projectId, String region, String clusterName)
+      throws IOException {
+
+    String myEndpoint = String.format("%s-dataproc.googleapis.com:443", region);
+
+    // Configure the settings for the cluster controller client
+    ClusterControllerSettings clusterControllerSettings =
+        ClusterControllerSettings.newBuilder().setEndpoint(myEndpoint).build();
+
+    // Create a cluster controller client with the configured settings
+    try (ClusterControllerClient clusterControllerClient = ClusterControllerClient.create(clusterControllerSettings)) {
+
+      // Configure the settings for our cluster
+      InstanceGroupConfig masterConfig = InstanceGroupConfig.newBuilder().setMachineTypeUri("n1-standard-1")
+          .setNumInstances(1).build();
+
+      InstanceGroupConfig workerConfig = InstanceGroupConfig.newBuilder().setMachineTypeUri("n1-standard-1")
+          .setNumInstances(2).build();
+
+      ClusterConfig clusterConfig = ClusterConfig.newBuilder().setMasterConfig(masterConfig)
+          .setWorkerConfig(workerConfig).build();
+
+      // Create the cluster object with the desired cluster config
+      Cluster cluster = Cluster.newBuilder().setClusterName(clusterName).setConfig(clusterConfig).build();
+
+      // Create the cluster
+      Cluster response = clusterControllerClient.createClusterAsync(projectId, region, cluster).get();
```

Comment:
Please separate out the request creation from the `.get` - We want it to be obvious to users that a future is being returned here.

Reply:
Do you feel strongly about this? I followed the convention in the doc- [(https://googleapis.dev/java/google-cloud-clients/latest/com/google/cloud/dataproc/v1/ClusterControllerClient.html#createClusterAsync-com.google.cloud.dataproc.v1.CreateClusterRequest-)]

Either way, I feel these should match

Reply:
Yes - creation of a future followed immediately by a `.get` is an anti-pattern. It should be clear that the `OperationFuture` is a long running operation and may block for an extended period of time.

Reply:
Ok, fixed + commited this in the sample and the test, and made a note to fix the javadoc to match this.